### ISSUE TITLE
Blog Post:  The Tension of High-Performance Computing: Reproducibility vs. Parallelization

### ISSUE THREAD

**Issue by AleKoure** [pharmaverse/blog#195]
### Blog Post

Balancing reproducibility and parallelization in data science presents a trade-off due to several factors. Managing consistent RNG states and process synchronization in parallel tasks is challenging, leading to potential inconsistencies. This trade-off exists because parallel computing aims to optimize performance and speed, often at the expense of reproducibility. The need to ensure reliable and reproducible results across different environments further complicates this balance, making it crucial to address these challenges to achieve high-performance computing without compromising data integrity. Tools like mirai in R can help by facilitating asynchronous computation, making it possible to reduce the tension between high-performance computing and reproducibility.

**Comment by bms63** [comment]
Hi @AleKoure - if this is written in a way that discusses challenges biostatisticians and statistical programmers face when working with large data, conduct simulations, modeling, etc then sounds a like a really interesting topic.  Also, assuming you are thinking about writing this post?

What do you think @manciniedoardo @StefanThoma @gigikenneth @kaz462

**Comment by manciniedoardo** [comment]
Absolutely! Also would want to tie in discussions around specific pharmaverse packages

**Comment by AleKoure** [comment]
Great! 
We had this discussion with @StefanThoma at useR! and he suggested opening an issue here.
Let's have some simple examples in base R and then reframe using e.g. admiral or any other suggestion you have.

**Comment by StefanThoma** [comment]
Hi @AleKoure 
When do you think you'll have a first draft ready?

**Comment by AleKoure** [comment]
Hi, I'll make a draft by the beginning of Sep, and then we can sync.
I'll use a simple simulation of drug efficacy and maybe some RNG, and
discuss around {mirai}.
If you have any additional inputs, suggestions, or specific points, please
feel free to share them :)

**Comment by StefanThoma** [comment]
Hi @AleKoure, 
How's the first draft coming along?

**Comment by AleKoure** [comment]
@StefanThoma Thanks for reaching out! 
I haven’t made much progress yet, but I have some simple examples (logrx + mirai) and text ready. 
I was able to rebuild the Docker image, and I plan to glue these together in a .qmd file, aiming 
to submit the PR within the next 1-2 days.

**Comment by AleKoure** [comment]
Breaking news: My merge request is done! :smile_cat: 

Thank you @manciniedoardo for already taking the time to comment on the draft version! 🚀 Much appreciated!

I couldn’t tag the reviewers since they’ve already been assigned.

@StefanThoma, @sikokuchuo if you have the time PTAL! since this started from our discussions at useR!

**Comment by shikokuchuo** [comment]
Congratulations @AleKoure with the PR - I saw that it was a mammoth task!

It's nice that you're featuring mirai in this blog post as well. I just released mirai 1.3.0 today and the dispatcher argument is now `dispatcher = "none"` rather than `dispatcher = FALSE`, although the latter still works. This is as I added threaded dispatcher in case you're wondering - this is a more efficient version of dispatcher - it's not the default yet as it needs more real-world testing.
